### PR TITLE
Adapt kibana integrations to public interfaces of saved objects for 7.11/7.12

### PR DESCRIPTION
**Pull request by frankeros**
| Related to |  #3213 |
|--|--|

Hi team, 
this resolves:
- Wazuh app doesn't up properly in dev mode with Kibana 7.11 and Kibana 7.12

The changes affect to kibana-discover and kibana-vis.
Testing:
Check some modules for Dashboard and Events works properly in: 
- [ ] Kibana 7.10.2 with ODFE and basic
- [ ] Kibana 7.11.2 basic
- [ ] Kibana 7.12.1 basic

### CODE REVIEW

File: public/services/resolves/get-saved-search.js [hunk 1]
```diff
@@ -11,59 +11,65 @@
...
+            },
+          });
 
-        /* redirectWhenMissing({
+          /* redirectWhenMissing({
           search: '/discover',
           'index-pattern':
             '/management/kibana/objects/savedSearches/' +
             $route.current.params.id
         }) */
```
Comment (gabiwassan) [inline public/services/resolves/get-saved-search.js:69]: Could you remove this block commented code :)

**Review by gabiwassan — APPROVED** [review]
LGTM 👍🏼

**Review by mpRegalado — APPROVED** [review]
**CR:** LGTM! :heavy_check_mark: 
**Testing:** LGTM! :heavy_check_mark: 
For each test I checked that:

- The app passes the health checks.
- You can visit the security events page, as to check that the file `public/services/resolves/get-saved-search.js` is executed correctly
- You can visit the agents overview page to check the execution of the `public/kibana-integrations/visualizations/_saved_vis.ts` and `public/kibana-integrations/kibana-vis.js` files.

On all the four enviroments mentioned in the PR, all my tests looked ok.

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark: 
Test: :heavy_check_mark: 

LGTM!

**Pull request merged**